core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTests.java

```diff
@@ -79,18 +84,23 @@ public FieldSortBuilder randomFieldSortBuilder() {
         if (randomBoolean()) {
             builder.sortMode(randomFrom(SortMode.values()));
         }
-
         if (randomBoolean()) {
             builder.setNestedSort(createRandomNestedSort(3));
         }
-
+        // the following are alternative ways to setNestedSort for nested sorting
+        if (randomBoolean()) {
+            builder.setNestedFilter(randomNestedFilter());
```

Comment:
Not related to tests but this function should be deprecated.

Reply:
Not sure if this was omitted on purpose, but I added deprecation annotations to the setter and getter